src/app/data_harvester/disks/freebsd.rs

```diff
@@ -25,14 +29,32 @@ struct FileSystem {
 }
 
 pub fn get_io_usage() -> error::Result<IoHarvest> {
-    let io_harvest = get_disk_info().map(|storage_system_information| {
-        storage_system_information
-            .filesystem
-            .into_iter()
-            .map(|disk| (disk.name, None))
-            .collect()
-    })?;
+    #[allow(unused_mut)]
+    let mut io_harvest: HashMap<String, Option<IoData>> =
+        get_disk_info().map(|storage_system_information| {
+            storage_system_information
+                .filesystem
+                .into_iter()
+                .map(|disk| (disk.name, None))
+                .collect()
+        })?;
```

Comment:
Hm, might be worth changing this to not fail now if `get_disk_info()` fails via the `?` if the zfs part might also provide more info. Feel free to skip this, I can also look at it.

Reply:
I would change this to be more infallible... but if `get_disk_info` fails then the `IoCounters` will not be used regardless?
In the `data_farmer` the `device_name` we use for the `checked_name` to get the `io_device` information out of the `IoHarvest` map [comes from ](https://github.com/ClementTsang/bottom/blob/ccc70915293607a001816238dcd3e0986e05bd26/src/app/data_farmer.rs#L341)`DiskHarvest`. 
`DiskHarvest` on freebsd [uses the same](https://github.com/ClementTsang/bottom/blob/ccc70915293607a001816238dcd3e0986e05bd26/src/app/data_harvester/disks/freebsd.rs#L42) `get_disk_info` as `get_io_usage`. 
So if `get_disk_info` is failing the Io information will not be used as the key will not be retrieved and the values turned into an `io_label`... it may require another pass to utilize the `IoHarvest` information regardless of `DiskHarvest`?